skopt/space.py

```diff
@@ -351,22 +357,12 @@ def transform(self, X):
             The transformed samples.
         """
         # Pack by dimension
-        columns = []
-        for dim in self.dimensions:
-            columns.append([])
-
-        for i in range(len(X)):
-            for j in range(self.n_dims):
-                columns[j].append(X[i][j])
-
-        # Transform
-        for j in range(self.n_dims):
-            columns[j] = self.dimensions[j].transform(columns[j])
-
-        # Repack as an array
-        Xt = np.hstack([np.asarray(c).reshape((len(X), -1)) for c in columns])
+        X = np.asarray(X, dtype=object)
+        X_transform = [
+            self.dimensions[j].transform(X[:, j]).reshape(len(X), -1)
+            for j in range(self.n_dims)]
 
-        return Xt
+        return np.hstack(X_transform)
```

Comment:
Then to be sure that everything is under control, shall we cast here? i.e. return `np.hstack(X_transform).astype(np.float)`?